### PR TITLE
Top-left progress panel - initial take

### PR DESCRIPTION
**Pull request by jinzhong2**
1) changed slider to show time progress
2) changed the layout and speed+/- icons
3) changed speed to show actual speed
todo: drag/click slider to the right needs to set new simulation time, and redraw the window
todo: drag/click slider to the left  needs to trigger dialogue (reset to 0, or cancel action)
todo: missing pause button icon
todo: remove the background circle of all buttons
todo: redo edit/jump time panel
todo: maybe make the panel wider to accommodate sunrise and sunset under the slider bar.

### CODE REVIEW

File: game/src/common/speed.rs [hunk 1]
```diff
@@ -171,17 +166,16 @@ impl SpeedControls {
                     let now = Instant::now();
                     self.state = State::Running {
                         last_step: now,
-                        speed_description: "...".to_string(),
                         last_measurement: now,
                         last_measurement_sim: current_sim_time,
                     };
+                    self.slider.set_percent(ctx, current_sim_time.inner_seconds()/86400.0);
```
Comment (dabreegster) [inline game/src/common/speed.rs]: Did current_sim_time / Duration::END_OF_DAY not work?

File: game/src/common/speed.rs [hunk 2]
```diff
@@ -149,19 +151,12 @@ impl SpeedControls {
...
-            );
-        } else if self.slow_down_btn.clicked() && desired_speed != 0.0 {
-            self.slider.set_percent(
-                ctx,
-                (self.slider.get_percent() - ADJUST_SPEED_PERCENT).max(0.0),
-            );
+        if self.speed_up_btn.clicked() && self.speed_actual != self.speed_cap {
+            self.speed_actual += self.speed_cap * ADJUST_SPEED_PERCENT;
+        } else if self.slow_down_btn.clicked() && self.speed_actual != 0.0 {
+            self.speed_actual -= self.speed_cap * ADJUST_SPEED_PERCENT;
```
Comment (dabreegster) [inline game/src/common/speed.rs]: self.speed_cap is the highest speed setting that should be possible for people to set -- 600x normally. I'm not sure multiplying by it makes sense. Also, it's possible to request negative speed here -- that's what the .max(0.0) was guarding against previously

File: game/src/common/speed.rs (hunk 1 above)
Comment (jinzhong2) [reply game/src/common/speed.rs]: Doesn't work. current_sim_time is f64, and Duration::END_OF_DAY is not f64.
A question regarding to data type, why all the dimensions used are f64? width/height are all integers.

File: game/src/common/speed.rs (hunk 2 above)
Comment (jinzhong2) [reply game/src/common/speed.rs]: top speed_cap * ADJUST_SPEED_PERCENT gives the speed change.  PERCENT is 1% (0.01), so the total change is 6. Can certainly change it to less, like 0.002, so give the increment of about 1. Have you tried the code? Works well actually.

File: game/src/common/speed.rs (hunk 1 above)
Comment (dabreegster) [reply game/src/common/speed.rs]: I tried `self.slider.set_percent(ctx, current_sim_time / Duration::END_OF_DAY);` and it seemed to work.
I don't have a consistent story for units yet. Maybe screen dimensions and more should be integers; I need to audit all of the geometry and GUI types and rethink some of them.

File: game/src/common/speed.rs (hunk 2 above)
Comment (dabreegster) [reply game/src/common/speed.rs]: Clicking the slow down button a few times, I can request negative speeds: https://imgur.com/HJ6711I

**Comment by dabreegster** [comment]
I made a new branch, so we can merge this into that and keep working. Thanks for the PR! :)

**Pull request merged**